flocker/control/test/test_persistence.py

```diff
@@ -249,3 +303,88 @@ def test_complex_keys(self):
                                manifestations={}, paths={},
                                devices={uuid4(): FilePath(b"/tmp")})
         self.assertEqual(node_state, wire_decode(wire_encode(node_state)))
+
+
+class FakeMigration(object):
+    """
+    A simple fake migration class, used to test ``migrate_configuration``.
+    """
+    @classmethod
+    def configuration_v1_v2(cls, config):
+        config = json.loads(config)
+        if config['version'] != 1:
+            raise ConfigurationMigrationError(
+                "Supplied configuration was not a valid v1 config."
+            )
+        return json.dumps({"version": 2, "configuration": "fake"})
+
+    @classmethod
+    def configuration_v2_v3(cls, config):
+        config = json.loads(config)
+        if config['version'] != 2:
+            raise ConfigurationMigrationError(
+                "Supplied configuration was not a valid v2 config."
+            )
+        return json.dumps({"version": 3, "configuration": "fake"})
+
+
+class ConfigurationMigrationTests(SynchronousTestCase):
+    """
+    Tests for ``ConfigurationMigration`` and ``migrate_configuration``.
```

Comment:
This test suite only demonstrates that the version number gets bigger.  It doesn't demonstrate that the content or structure of the configuration can be changed.